utilities/Microsoft.Quantum.Katas/KataMagic.cs

```diff
@@ -178,8 +186,17 @@ public virtual bool Simulate(OperationInfo test, OperationInfo userAnswer, IChan
         /// It does this by finding another operation with the same name as the `userAnswer` but in the 
         /// test's namespace
         /// </summary>
-        public virtual OperationInfo FindSkeletonAnswer(OperationInfo test, OperationInfo userAnswer) =>
-            Resolver.Resolve($"{test.Header.QualifiedName.Namespace.Value}.{userAnswer.FullName}");
+        public virtual OperationInfo FindSkeletonAnswer(OperationInfo test, OperationInfo userAnswer)
+        {
+            var skeletonAnswer = Resolver.Resolve($"{test.Header.QualifiedName.Namespace.Value}.{userAnswer.FullName}");
+            Console.Out.WriteLine($"Resolved {userAnswer.FullName} to {skeletonAnswer}");
+            if (skeletonAnswer != null)
+            {
+                // Remember the last user answer for this task
+                AllAnswers[skeletonAnswer] = userAnswer;
```

Comment:
Have you confirmed that `OperationInfo` works well as key for dictionary? I think it does, but it would be nice to make sure that if you run the same cell multiple times indeed the latest userAnswer is the one used.

Reply:
Yes, I verified it using the pair of Alice/Bob strategies from #115 - the latest solution from each cell is used.